catchup/service_test.go

```diff
@@ -535,14 +533,16 @@ func helperTestOnSwitchToUnSupportedProtocol(
 	config := defaultConfig
 	config.CatchupParallelBlocks = 2
 
-	remote, _, blk, err := buildTestLedger(t, bookkeeping.Block{}) //mRemote.blocks[0])
+	block1 := mRemote.blocks[1]
```

Comment:
I like this the previous way because in this way it could erroneously use block1 or mRemote.blocks[1] after buildTestLedger, but it's pass by value not pass by reference. I want to avoid trying to use the unmodified blank block and thinking we're getting something else.

Reply:
What do you mean by?
> unmodified blank block


mRemote.blocks[1] will have the NextProtocolSwitchOn set according to the parameters passed to NextProtocolSwitchOn. That is why it is important to pick mRemote.blocks[1] instead of bookkeeping.Block{}.

Reply:
Your other comment said no change in behavior, just reorg, but this line is a change in behavior of what `buildTestLedger()` gets from `Block{}` to whatever `testingenvWithUpgrade()` may have put into `mRemote.blocks[1]`. Just checking that's intentional and there's no pass-by-value pass-by-reference mixup here.

Reply:
oh I see. sorry for the confusion I caused with that comment. 
That comment applies exclusively to the TestOnSwitchToUnSupportedProtocol function.